config/repository.go

```diff
@@ -0,0 +1,271 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"sync"
+
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/util"
+)
+
+const (
+	ConfigPrefix string = "config"
+	TenantPrefix string = "tenant-"
+)
+
+// Repository defines the required methods for the config repository.
+type Repository interface {
```

Comment:
In the next PR might create 2 different repositories as we will handle more gracefully code duplication.